### PR TITLE
Forks/enes/data authorization

### PR DESCRIPTION
**Pull request by mcarey1590**
Adds configuration hooks into AutoCrud to support resource level authorization
- CRUD endpoints
- Change Tracking endpoints
- Custom fields endpoints

Sample App updates to include authentication/authorization to show how resource authorization can be implemented with AutoCrud

Integration tests updates to include resource authorization checks

image

### CODE REVIEW

File: Firebend.AutoCrud.Web.Sample/DbContexts/DbFunctionsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System;
```
Comment (JacobAtchley) [inline Firebend.AutoCrud.Web.Sample/DbContexts/DbFunctionsExtensions.cs:1]: if these are not supported natively in EF Core now, would it be worth while to move them [here](https://github.com/firebend/auto-crud/tree/main/Firebend.AutoCrud.EntityFramework/CustomCommands) so they can be utilized by a wider audience?

**Review by JacobAtchley — CHANGES_REQUESTED** [review]
this is looking good! thanks guys! i just had a comment or two about the test 👍

File: Firebend.AutoCrud.Web.Sample/DbContexts/DbFunctionsExtensions.cs (hunk 1 above)
Comment (mcarey1590) [reply Firebend.AutoCrud.Web.Sample/DbContexts/DbFunctionsExtensions.cs:1]: yeah, sounds good

Comment (mcarey1590) [reply Firebend.AutoCrud.Web.Sample/DbContexts/DbFunctionsExtensions.cs]: Moved 52205f1

File: Firebend.AutoCrud.Tests/ChangeTracking/Web/Authorization/EntityChangeTrackingAuthorizationFilterTests.cs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using NUnit.Framework;
+
+namespace Firebend.AutoCrud.Tests.ChangeTracking.Web.Authorization;
+
+[TestFixture]
+public class EntityChangeTrackingAuthorizationFilterTests
```
[the remark this answers is not in the document]
Comment (enesccinar) [reply Firebend.AutoCrud.Tests/ChangeTracking/Web/Authorization/EntityChangeTrackingAuthorizationFilterTests.cs:26]: of course! updated: https://github.com/firebend/auto-crud/pull/90/commits/ac4ac90cad9209945ce6b8ef7152abf8c4cfab38

[the remark this answers is not in the document]
Comment (enesccinar) [reply Firebend.AutoCrud.Tests/ChangeTracking/Web/Authorization/EntityChangeTrackingAuthorizationFilterTests.cs:26]: image

**Pull request merged**